File: peacedesign/src/main/java/com/peacedesign/android/widget/list/singleChoice/SingleChoiceListView.java
package com.peacedesign.android.widget.list.singleChoice;

import android.content.Context;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.peacedesign.android.widget.list.base.BaseListItem;
import com.peacedesign.android.widget.list.base.BaseListView;
import com.peacedesign.android.widget.radio.PeaceRadioGroup;

public class SingleChoiceListView extends BaseListView {
    @Nullable
    private OnItemClickListener mSelectionChangeListener;

    public SingleChoiceListView(@NonNull Context context) {
        super(context);
    }

    @NonNull
    @Override
    protected ViewGroup createItemsContainer(@NonNull Context context) {
        PeaceRadioGroup radioGroup = new PeaceRadioGroup(context);
        radioGroup.setOnCheckedChangedListener((button, checkedId) -> {
            if (mSelectionChangeListener != null) {
                BaseListItem item = (BaseListItem) button.getTag();
                if (item != null) {
                    mSelectionChangeListener.onItemClick(item);
                }
            }
        });
        return radioGroup;
    }

    @Override
    public void setOnItemClickListener(@NonNull OnItemClickListener listener) {
        mSelectionChangeListener = listener;
    }
}

File: app/src/main/java/com/quranapp/android/widgets/list/singleChoice/SingleChoiceListView.kt
package com.quranapp.android.widgets.list.singleChoice

import android.content.Context
import android.view.ViewGroup
import com.quranapp.android.widgets.list.base.BaseListItem
import com.quranapp.android.widgets.list.base.BaseListView
import com.quranapp.android.widgets.radio.PeaceRadioGroup

class SingleChoiceListView(context: Context) : BaseListView(context) {
    private var selectionChangeListener: OnItemClickListener? = null

    override fun createItemsContainer(context: Context): ViewGroup {
        val radioGroup = PeaceRadioGroup(context)
        radioGroup.onCheckChangedListener = { button, _ ->
            if (selectionChangeListener != null) {
                val item = button.tag as BaseListItem?
                if (item != null) {
                    selectionChangeListener!!.onItemClick(item)
                }
            }
        }
        return radioGroup
    }

    override fun setOnItemClickListener(listener: OnItemClickListener) {
        selectionChangeListener = listener
    }
}
